processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -341,7 +435,18 @@ public void advance()
                       if (Thread.interrupted()) {
                         throw new QueryInterruptedException(new InterruptedException());
                       }
-                      cursorOffset.increment();
+                      cursorOffsetHolder.get().increment();
+
+                      while (!isDone()) {
```

Comment:
I had a benchmark called `read` within https://github.com/jon-wei/druid/blob/twostage_filter_bench2/benchmarks/src/main/java/io/druid/benchmark/FilterPartitionBenchmark.java

that opens a cursor with no filters, with 750,000 rows and feeds each row to the benchmark blackhole, I think this would give a good estimate of cursor-only performance 

The time it took was about the same on master and with this patch

master:
```
FilterPartitionBenchmark.read                              750000     basic               avgt   50    87098.678 ±  2167.394  us/op
```

with patch:
```
FilterPartitionBenchmark.read                              750000     basic               avgt   50    86685.989 ±  2527.736  us/op
```